### DISCUSSION TITLE
Linebuffer synchronization

### DISCUSSION THREAD

**Discussion by 0xa000** [projf/projf-explore#142] [Q&A] [answered]
Hello,

I'm really enjoying your FPGA graphics tutorials! There is a detail framebuffers post that puzzled me. I *think* I understand it now, but I thought I'd ask to be sure. Maybe it'll help someone else as well.

So, the thing is, in the "Scaling" section, data is copied from the framebuffer to the linebuffer for each line. The data for the each pixel of a line has to be available in the linebuffer before it is read by the part that design that outputs pixel data to the display. It was not obvious to me this is arranged. I think what happens is that copying data from the framebuffer to the linebuffer starts at the start of the horizontal blanking, whereas reading data from the linebuffer starts at the first active pixel (sx == 0). (Or actually, slightly before that because of latency.) Is this correct? 

Thanks!

**Comment by WillGreen** [comment, marked as answer]
Yes, you've got it! Because blanking occurs before the start of active drawing, we can start loading the data into the linebuffer in plenty of time to draw the first pixel on the screen.

The blog post could explain it better.
